app/controllers/application_controller.rb

```diff
@@ -4,13 +4,24 @@ class ApplicationController < ActionController::Base
 
   rescue_from ActionController::InvalidAuthenticityToken, with: :session_expired
   rescue_from ActionController::RoutingError, with: :render_not_found
+  rescue_from GetIntoTeachingApiClient::ApiError, with: :handle_api_error
 
   def raise_not_found
     raise ActionController::RoutingError, "Not Found"
   end
 
 private
 
+  def handle_api_error(error)
+    render_too_many_requests && return if error.code == 429
+
+    raise
+  end
+
+  def render_too_many_requests
+    render template: "errors/too_many_requests", status: :too_many_requests
```

Comment:
This should specify the layout as well - I got bitten by this previously - template inheritance meant it defaulted to the wrong layout for certain pages if the descendant controller has the same name as a layout file.